### PR TITLE
[CARBONDATA-1323] Presto Optimization for Integration Layer

### PR DESCRIPTION
**Pull request by bhavya411**
Changes done to increase the performance of Presto Integration
- Removed Extra loops to save time on Object creation
- Implemented Lazy Blocks so that data in block in not initialized in first go.
- Implemented a new efficient dictionary decoder.

Build is verified
All test cases passing
Manually verified the data

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/575/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3170/

File: integration/presto/pom.xml [hunk 1]
```diff
@@ -228,6 +228,33 @@
             <skip>true</skip>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.scala-tools</groupId>
+          <artifactId>maven-scala-plugin</artifactId>
```
Comment (chenliang613) [inline integration/presto/pom.xml]: can you explain, why need add the plugin to pom file?

Comment (bhavya411) [inline integration/presto/pom.xml]: I have written the dictionary decoding in scala as it is more optimized and easier to understand, hence we have to add this plugin for compiling the scala code

**Pull request closed without merging**